src/main/java/org/zalando/nakadi/controller/NakadiController.java

```diff
@@ -0,0 +1,74 @@
+package org.zalando.nakadi.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.zalando.nakadi.domain.ItemsWrapper;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.FloodService;
+import org.zalando.nakadi.util.FeatureToggleService;
+
+import static org.zalando.nakadi.security.Client.FULL_ACCESS;
+
+@RestController
+@RequestMapping(value = "/nakadi")
+public class NakadiController {
+
+    private final FloodService floodService;
+    private final FeatureToggleService featureToggleService;
+
+    @Autowired
+    public NakadiController(final FloodService floodService,
+                            final FeatureToggleService featureToggleService) {
+        this.floodService = floodService;
+        this.featureToggleService = featureToggleService;
+    }
+
+    @RequestMapping(path = "/flooders", method = RequestMethod.GET)
+    public ResponseEntity<?> getFlooders(final Client client) {
+        if (client != FULL_ACCESS) {
```

Comment:
I think this is not a correct check. As I see in `ClientResolver`, one of the use-cases for getting FULL_ACCESS is when feature CHECK_APPLICATION_LEVEL_PERMISSIONS is not enabled. So when we deploy your code on live it will be possible for everyone to call these endpoints.